src/lib/prep.ml

```diff
@@ -200,16 +200,11 @@ module Prep = struct
     in
 
     let fn () =
-      let* result = Current_ocluster.Connection.run_job ~job build_job in
-      let* x = result
-        |> Result.fold ~error:(fun _ -> Error ()) ~ok:(fun _ -> Ok(([], []), []))
-        |> Misc.fold_logs build_job (Misc.with_error_check extract_hashes)
+      let** _ = Current_ocluster.Connection.run_job ~job build_job in
+      let* x =
+        Misc.fold_logs build_job extract_hashes (([],[]), [])
       in
-      match x with
-      | Ok (Ok x) -> Lwt.return_ok x
-      | Ok (Error ()) -> Lwt.return_error (`Msg "Compile: failed to run job")
-      | Error s -> Lwt.return_error s
-
+      Lwt_result.lift x
```

Comment:
I've made this a lot simpler.

1. Run the job
2. Regardless of the outcome of the job, run fold_logs to collect retriable errors

Thus, the retry machinery has the information it needs to return the results of fold_logs ... or retry.

Reply:
Sounds good to me !